addons/website_profile/controllers/main.py

```diff
@@ -37,13 +37,17 @@ def _check_avatar_access(self, user_id, **post):
 
     def _check_user_profile_access(self, user_id):
         user_sudo = request.env['res.users'].sudo().browse(user_id)
+        result = {'denial': False, 'not_found': False, 'user': user_sudo}
```

Comment:
As this function is used only below, could we instead use 3 variables and return a tuple?
This makes it generally more readable, doesn't it?
```suggestion
        denial_reason = False
        user_not_found = False
        ...
        return user_sudo, denial_reason, user_not_found
```